lib/liquid.lua

```diff
@@ -2547,28 +2549,39 @@ end
 -------------------------------------------------------------ParserContext end ---------------------------------------------------------
 -------------------------------------------------------------FileSystem begin ---------------------------------------------------------
 -- local FileSystem = {}
-function FileSystem:new( location )
+function FileSystem:new( get, error_handler )
     -- body
     local instance = {}
     setmetatable(instance, {__index = FileSystem})
-    instance.location = location
+    instance.get = get
     instance.text = nil
+    instance.error_handler = error_handler
     return instance
 end
 --
-function FileSystem:genertic_get( ... )
+function FileSystem:genertic_get( location )
+    local error_handler = self.error_handler
     -- body
     if self.get and type(self.get) == "function" then
-        local status, err = pcall(self.get, self.location)
-        if status then
-            return err
+        local ok, ret = pcall(self.get, location)
+
+        if not ret then
+            return error_handler(location, "cannot render empty template" )
+	      end
+
+        if ok then
+            return ret
         else
-            error("get template: " .. location .. " fail by user self defined get method" .. tostring(err))
+            return error_handler(location, "fail by user self defined get method: " .. tostring(ret))
```

Comment:
Will the  the interface of  `error_handler(location, error_msg_string)` too simple?
how about using `error_handler(location, error_msg_string, error_code)`?  The `error_code` will be much exact and easy to disguish the type of error, but it also means we should maintain a list of error codes.

any suggestions?